src/miner.h

```diff
@@ -203,6 +203,6 @@ void IncrementExtraNonce(CBlock* pblock, const CBlockIndex* pindexPrev, unsigned
 int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParams, const CBlockIndex* pindexPrev);
 
 /** Update an old GenerateCoinbaseCommitment from CreateNewBlock after the block txs have changed */
-void RegenerateCommitments(CBlock& block, CBlockIndex* prev_block);
+void RegenerateCommitments(CBlock& block, ChainstateManager& chainman);
```

Comment:
Perhaps just pass in a `BlockManager&` here?

Reply:
This is the 4th time the same function gets changed around. I think we should decide what to settle on, and ideally provide a rationale for it. 

https://github.com/bitcoin/bitcoin/pull/21391#discussion_r615237925

Reply:
I think fundamentally the issue here is with `generateblock()`, which calls into miner.cpp to construct the block, then changes the transactions in that block, and then calls back into miner.cpp to fix up the coinbase commitment. It'd be a much simpler interface if `generateblock()` just passed in a const ref to the vector of transactions it wants to include in its first call to miner.cpp.

I'm going to mark this comment as resolved. The mining code is a bit of a mess, and can be cleaned up in other PRs.